File: Ejercicios/Decorator/After/Decorator.Starbuzz/Soy.cs
namespace Decorator.Starbuzz
{
	public class Soy : CondimentDecorator
	{
		Beverage beverage;
		
		public Soy(Beverage beverage)
		{
			this.beverage = beverage;
		}

		public override string GetDescription()
		{
			return this.beverage.GetDescription() + ", Soy";
		}

		public override double Cost()
		{
			return .15 + this.beverage.Cost();
		}
	}
}

File: Ejercicios/2. Starbuzz (Beverages)/Java/After/src/main/starbuzz/Soy.java
package starbuzz;

public class Soy extends CondimentDecorator {
	Beverage beverage;

	public Soy(Beverage beverage) {
		this.beverage = beverage;
	}
	
	public double cost() {
		return .15 + beverage.cost();
	}
}
